src/main/scala/mesosphere/mesos/Constraints.scala

```diff
@@ -102,27 +116,31 @@ object Constraints {
             // All running tasks should have the matching attribute
             matches.size == tasks.size
         case Operator.GROUP_BY =>
-          val groupFunc = (task: Task) =>
-            task.agentInfo.attributes
-              .find(_.getName == field)
-              .map(getValueString(_))
           checkGroupBy(getValueString(attr.get), groupFunc)
-        case Operator.LIKE =>
-          if (value.nonEmpty) {
-            getValueString(attr.get).matches(value)
-          }
-          else {
-            log.warn("Error, value is required for LIKE operation")
-            false
-          }
-        case Operator.UNLIKE =>
-          if (value.nonEmpty) {
-            !getValueString(attr.get).matches(value)
-          }
-          else {
-            log.warn("Error, value is required for UNLIKE operation")
-            false
-          }
+        case Operator.MAX_PER =>
+          checkMaxPer(offer.getHostname, value.toInt, groupFunc)
+        case Operator.LIKE   => checkLike
```

Comment:
checkLike and checkLike helpers could be unified (since they are exactly the same), e.g:

`case op@(_:Operator.LIKE | _:Operator.UNLIKE) => checkLike(op.getName)
}

def checkLike(name: String) { ... } ` and log the warning with the name of the operator.